theano/tensor/tests/test_opt.py

```diff
@@ -3727,6 +3725,22 @@ def test_xor(self):
         self.assert_eqs_const(f, 0)
 
 
+    def test_stacktrace(self):
+        mode = theano.compile.get_default_mode().including(
+            'local_useless_elemwise_comparison')
+
+        x = T.vector('x', dtype=config.floatX)
+        f = theano.function([x], T.gt(x, x), mode=mode)
+        self.assertTrue(check_stack_trace(f, ops_to_check='last'))
+
+        f = theano.function([x], T.le(x, x), mode=mode)
+        self.assertTrue(check_stack_trace(f, ops_to_check='last'))
+
+        # Julian: I tried testing the stack trace for a bunch of different
+        # functions, including maximum and shapes, but other opts remove
+        # the stack traces in this case.
```

Comment:
I tried adding this test:

    mode = theano.compile.get_default_mode().including(
            'local_useless_elemwise_comparison')
    f = theano.function([x], T.minimum(Shape_i(0)(x), 0), mode=mode)
    self.assertTrue(check_stack_trace(f, ops_to_check='all'))

However, it doesn't work. The stack trace gets removed somewhere along the way.

The unoptimized graph looks like this:

    Elemwise{minimum,no_inplace} [id A] ''   
     |Shape_i{0} [id B] ''   
     | |x [id C]
     |TensorConstant{0} [id D]

The optimized graph looks like this:

    DeepCopyOp [id A] ''   0
     |TensorConstant{0} [id B]

I'm not sure what's going on, but "local_useless_elemwise_comparison" definitely does copy over the stack trace.